### PR TITLE
79 card

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Creates Vertical Card Component
- Creates Horizontal Card Variant
- Adds start to modular spacing mixins
- Adds start to box shadows
- Makes improvements to button component
- Removes and updates input.html to input.twig
- Has some dependencies on typography mixins
- Will need to refactor modular and shadow mixins

# Needed By (Date)
- End of the week?

# Urgency
- Medium-low

# Steps to Test

1. Read through the code and leave comments to approach
2. Check out this branch locally
3. Compile style guide
4. Review Card component in browser for look and feel
5. Review Card component in browser for accessibility and semantics
6. Review Card component in browser for responsiveness

# Associated Issues and/or People
- #79 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: core/scss/components/card/_card.scss [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    @media #{$breakpoint-xl} {
+      margin-bottom: 19px;
+    }
+
+  }
+
+}
+
+// Call to action link.
+.su-card__cta-link {
```
Comment (JBCSU) [inline core/scss/components/card/_card.scss]: Please replace with an include of (new) `action` variant of link

File: core/scss/utilities/mixins/modular-scales/_modular-margin.scss [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    margin: map-get($modular-spacing, 'md');
+  }
+
+  margin: map-get($modular-spacing, 'xs');
+}
+
+// Function to get value.
+@function modular-margin-get-value($breakpoint) {
+  @return map-get($modular-spacing, $breakpoint);
+}
```
Comment (JBCSU) [inline core/scss/utilities/mixins/modular-scales/_modular-margin.scss]: Nice!!!

File: core/scss/utilities/mixins/shadows/_box-shadow.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  'shallow': '0px 3px 6px rgba(0, 0, 0, 0.13), 0px 3px 6px rgba(0, 0, 0, 0.1)',
+  'medium': '0px 10px 20px rgba(0, 0, 0, 0.15), 0px 6px 6px rgba(0, 0, 0, 0.2)',
+  'medium-deep': '0px 14px 28px rgba(0, 0, 0, 0.2), 0px 5px 10px rgba(0, 0, 0, 0.22)',
+  'deep': '0px 19px 38px rgba(0, 0, 0, 0.25), 0px 5px 12px rgba(0, 0, 0, 0.18)'
+) !default;
+
+@mixin box-shadow($depth: 'flat', $stroke: '1px solid #C6C6C6') {
+  border: #{$stroke};
+  box-shadow: #{map-get($box-shadow-depth, $depth)};
+}
```
Comment (JBCSU) [inline core/scss/utilities/mixins/shadows/_box-shadow.scss]: Also NICE!!!

File: core/templates/components/card/card.twig [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    {% endif %}
+  {% endblock %}
+
+  {# An open field for body content #}
+  {{ card_body }}
+
+  {# CTA Super Headline #}
+  {% if card_cta_link is not empty %}
+    <a {{ card_cta_attributes }} href="{{ card_cta_link }}" class="su-card__cta-link">{{ card_cta_label }}</a>
+  {% endif %}
```
Comment (JBCSU) [inline core/templates/components/card/card.twig]: Please replace `card_cta_link` with include of (new) `action` variant of a link. Also the `{# CTA Super Headline #}` comment is probably not what you want. ;)

File: core/templates/core/grid/grid.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {
-
+  "classes": ".su-width-"
```
Comment (JBCSU) [inline core/templates/core/grid/grid.json]: What is this for?

File: core/templates/elements/input/input.twig [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    <li>
+      <input id="etchemendy" type="radio" name="stanford-provosts-2" value="etchemendy">
+      <label for="etchemendy">John W. Etchemendy</label>
+    </li>
+    <li>
+      <input id="drell" type="radio" name="stanford-provosts-3" value="drell">
+      <label for="drell">Persis Drell</label>
+    </li>
+  </ul>
+</fieldset>
```
Comment (JBCSU) [inline core/templates/elements/input/input.twig]: Is this related to the card component???

File: core/scss/components/card/_card.scss [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  display: block;
+  color: $color-bright-blue;
+  font-size: 16px;
+  text-decoration: none;
+}
```
Comment (JBCSU) [inline core/scss/components/card/_card.scss]: Unnecessary after using the (new) `action` variant of a link?

File: core/templates/components/card/card.twig [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+<div{{ attributes }} class="su-card {{ modifier_class }}">
+
+  {# Card Image #}
+  {% if card_image is not empty %}
+    {{ card_image }}
+  {% endif %}
+
+  <div class="su-card__contents">
+
+  {# CTA Super Headline #}
```
Comment (JBCSU) [inline core/templates/components/card/card.twig]: In these comments, you probably want to replace 'CTA' with 'Card'

File: core/scss/elements/link/_link.scss [hunk 9]
```diff
@@ -5,21 +5,10 @@
...
-// Markup: ../../templates/elements/link/link.twig
+// Markup: ../templates/elements/link/link.twig
 //
 // Style guide: Elements.Link
 //
 a {
   @include link;
 }
-
-.su-external-link {
```
Comment (sherakama) [inline core/scss/elements/link/_link.scss]: I removed these in order to move them up into the link component with the other variants.

File: core/scss/components/card/_card.scss (hunk 7 above)
Comment (sherakama) [reply core/scss/components/card/_card.scss]: This was removed.

File: core/templates/components/card/card.twig (hunk 8 above)
Comment (sherakama) [reply core/templates/components/card/card.twig]: This has been changed.

File: core/templates/components/card/card.twig (hunk 4 above)
Comment (sherakama) [reply core/templates/components/card/card.twig]: This has been changed to use a link component now.

**Comment by sherakama** [comment]
@JBCSU @josephgknox 

Ready for the second review.

File: core/scss/components/card/_card.scss [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+
+// Card contents spacing.
+.su-card__contents {
+  @include modular-margin;
+
+  // Ensure no bottom margins on last item.
+  *:last-child {
+    margin-bottom: 0;
```
Comment (josephgknox) [inline core/scss/components/card/_card.scss]: Can we use the Bourbon margin mixin here for consistency?:

```suggestion
    @include margin(null null 0);
```

File: core/scss/components/card/_card.scss [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  // TODO: Refactor this to a mixin when typo headings available.
+  > h5 {
+    font-size: 13px;
+    font-weight: bold;
+    line-height: 17px;
+    margin-bottom: 11px;
+    margin-top: 0;
```
Comment (josephgknox) [inline core/scss/components/card/_card.scss]: See previous related comment on margin mixin. These two lines can be:

```suggestion
    @include margin(0 null 11px);
```

File: core/scss/components/card/_card.scss [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+  }
+
+  // TODO: Refactor this to a mixin when typo headings available.
+  > h3 {
+    font-size: 25px;
+    font-weight: bold;
+    line-height: 31px;
+    margin-bottom: 18px;
+    margin-top: 0;
```
Comment (josephgknox) [inline core/scss/components/card/_card.scss]: See previous related comment on margin mixin. These two lines can be:

```suggestion
    @include margin(0 null 18px);
```

File: core/scss/components/card/_card.scss [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+    @media #{$breakpoint-xl} {
+      margin-bottom: 19px;
```
Comment (josephgknox) [inline core/scss/components/card/_card.scss]: See previous related comment on margin mixin. This could be:

```suggestion
    @include margin(null null 19px);
```

File: core/scss/utilities/mixins/link/_more-link.scss [hunk 14]
```diff
@@ -1,21 +1,24 @@
...
 // Style guide: Mixins.Link.more-link
 //
-@mixin more-link($more-link, $more-link-hover) {
+@mixin more-link($color: $color-bright-red, $hover: $color-black) {
   &::after {
-    color: $color-bright-red;
+    color: $color;
     content: '\00BB';
     display: inline-block;
     margin-left: 4px;
```
Comment (josephgknox) [inline core/scss/utilities/mixins/link/_more-link.scss]: See previous related comment on margin mixin. This could be:

```suggestion
    @include margin(null null null 4px);
```
